packages/blockchain-extension/extension/commands/addEnvironmentCommand.ts

```diff
@@ -286,4 +241,128 @@ export async function addEnvironment(): Promise<void> {
+                        // This indicates a certificate verification is needed. Ask the user if they wish to proceed insecurely or cancel.
+                        const options: IBlockchainQuickPickItem<string>[] = [{ label: UserInputUtil.CONNECT_NO_CA_CERT_CHAIN, data: UserInputUtil.CONNECT_NO_CA_CERT_CHAIN }, { label: UserInputUtil.CANCEL_NO_CERT_CHAIN, data: UserInputUtil.CANCEL_NO_CERT_CHAIN, description: UserInputUtil.CANCEL_NO_CERT_CHAIN_DESCRIPTION }];
+                        const certificateUsage: IBlockchainQuickPickItem<string> = await UserInputUtil.showQuickPickItem('Unable to perform certificate verification. Please choose how to proceed', options) as IBlockchainQuickPickItem<string>;
+                        if (!certificateUsage || certificateUsage.label === UserInputUtil.CANCEL_NO_CERT_CHAIN) {
+                            return;
+                        }
+                    } else {
+                        // There is a connection problem other than certificate related. Throw error.
+                        throw new Error(`Unable to connect to the IBM Blockchain Platform network: ${errorWithoutAuth.message}`);
+                    }
+                }
+            }
+        }
+        // We will now atempt to connect using key and secret. If it fails, there is a problem with the key and secret provided.
+        try {
+            requestOptions.auth = { username: userAuth1, password: userAuth2 };
+            await Axios.get(api, requestOptions);
+        } catch (errorConnecting) {
+            throw new Error(`Problem detected with the authentication information provided: ${errorConnecting.message}`);
+        }
+        // Securely store API key and secret
+        try {
+            await keytar.setPassword('blockchain-vscode-ext', url, `${userAuth1}:${userAuth2}:${requestOptions.httpsAgent.options.rejectUnauthorized}`);
+        } catch (errorStorePass) {
+            throw new Error(`Unable to store the required credentials: ${errorStorePass.message}`);
+        }
+
+        return url;
+    }
+
+    async function getOpsToolsAccessInfoSaaS(): Promise<string[]> {
+        const ibpResources: IBlockchainQuickPickItem<string>[] = [];
+        const accessToken: string = await ExtensionsInteractionUtil.cloudAccountGetAccessToken(true);
+        if (!accessToken) {
+            return;
+        }
+        const requestOptions: any = { headers: { Authorization: `Bearer ${accessToken}` } };
+        const baseUrl: string = 'https://resource-controller.cloud.ibm.com';
+        let resourcesUrl: string = `${baseUrl}/v2/resource_instances`;
+        while (resourcesUrl) {
+            const response: any = await Axios.get(resourcesUrl.toString(), requestOptions);
+
+            for (const resource of response.data.resources) {
+                if ( resource.resource_plan_id === 'blockchain-standard' ) {
```

Comment:
What is the `resource_plan_id`? Does it come from IBP? Do we not want to support other resource_plan_id's?

Reply:
It's what identifies the resource as an IBM Blockchain platform, as opposed to any other resource in IBM cloud. We don't want to support any other.